tf2_ros_py/tf2_ros/buffer.py

```diff
@@ -94,6 +104,10 @@ def __get_frames(
     ) -> FrameGraphSrvResponse:
         return FrameGraph.Response(frame_yaml=self.all_frames_as_yaml())
 
+    def time_jump_callback(time_jump: TimeJump):
+        rclpy.logging.get_logger("tf2_buffer").warning("Detected jump back in time. Clearing tf buffer.")
+        self.clear()
```

Comment:
@erichlf @clalancette  I'd like to use this functionality, so I gave this a monkey patch test drive on humble but didn't manage to get the callback to trigger, alas. (when my rosbag looped).

Quick debugging: it doesn't seem to get _unregistered_, which is good, but the callback_shim doesn't seem to get called. 

This method doesn't look like it captures `self` so even if it were triggered I'm not sure this behaves how expected anymore (presumably it was working at some point but I saw there was refactoring during the PR).

Might be in need of some test coverage before the feature gets released :slightly_smiling_face:

Reply:
OH and I'm willing to admit that there's a slew of more complex reasons why I can't get it to trigger on my side (like, for example, monkey patching onto humble :laughing: ). But I'm hoping that the comment is helpful for rolling development nonetheless! :bow: